### PR TITLE
Limit number of concurrent reads/writes in TransferServer to num_xmits

### PR DESCRIPTION
**Pull request by nichhk**
testConcurrentRead in ReadWriteTest is not a great test because it does not assert that the number of transmits <= num_xmits, but I looked at the logging to make sure that dn->getNumXmits() is never greater than num_xmits. 

The test also depends on the first two reads finishing in <1min, because the other two readers will timeout otherwise. But it hasn't failed yet... 

@pelmers @srdieffenbach

### CODE REVIEW

**Comment by efrederic** [comment]
there are a lot of spacing issues here

File: rice-datanode/source/data_transfer_server.cc [hunk 1]
```diff
@@ -15,9 +15,9 @@ using asio::ip::tcp;
 using namespace hadoop::hdfs;
 
 // Default from CommonConfigurationKeysPublic.java#IO_FILE_BUFFER_SIZE_DEFAULT
-const size_t PACKET_PAYLOAD_BYTES = 4096;
+const size_t PACKET_PAYLOAD_BYTES = 4096 * 4;
```
Comment (efrederic) [inline rice-datanode/source/data_transfer_server.cc:18]: why are we increasing the packet size?

Comment (nichhk) [reply rice-datanode/source/data_transfer_server.cc:18]: It's a good amount slower at 4096.

Comment (Vegetable26) [reply rice-datanode/source/data_transfer_server.cc:18]: I noticed significant speedups after this change too.

**Pull request merged**